### PR TITLE
Crossed()/Uncrossed() consistency pass.

### LINKED ISSUE #3378: Multiple tapes affects ghosts
#### Description of issue

Engineering tapes affects ghosts, not characters

#### Difference between expected and actual behavior

As an engineer I expect to see message "you are supposed to cross eng tape" as well as "you are not for sec tape" I don't get this i game, but as a ghost only.

#### Steps to reproduce

Cross multiple tape types with multiple characters with different access
Do the same process with ghost of the character

#### Specific information for locating
Testing room: Bridge starboard storage

#### Length of time in which bug has been known to occur
24 September 2023, 2am UTC

#### Client version, Server revision & Game ID
Client Version: 514
Server Revision: 88ae4d3c8e33b4f51fbef970a1890596fd020a10 - staging - 2023-09-24
Game ID: cpX-acgh
Current map: Tradeship Ocelot Alpha

#### Issue bingo
- [x] Issue could be reproduced at least once
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Standardizes argument to `Crossed()`/`Uncrossed()`.
- Generally cleans up overrides.
- Fixes a few places where the crosser is insufficiently checked.
- Fixes #3378,
- Broke some vehicle and mulebot logic out into an /atom/movable proc (`crossed_mob()`).

## Why and what will this PR improve
Fixes a number of bugs, general code quality.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

File: code/game/objects/items/weapons/clown_items.dm [hunk 1]
```diff
@@ -7,10 +7,12 @@
 /*
  * Banana Peals
  */
-/obj/item/bananapeel/Crossed(var/atom/movable/AM)
-	if (istype(AM, /mob/living))
-		var/mob/living/M = AM
-		M.slip("the [src.name]", 4)
+/obj/item/bananapeel/Crossed(atom/movable/AM)
+	if(isliving(AM))
```
Comment (quardbreak) [inline code/game/objects/items/weapons/clown_items.dm]: Maybe `if(!isliving(AM))`?

Comment (MistakeNot4892) [reply code/game/objects/items/weapons/clown_items.dm]: Yeah that's a typo, thanks.

**Review by comma — APPROVED** [review]

File: code/game/objects/effects/decals/Cleanable/humans.dm [hunk 2]
```diff
@@ -88,30 +88,29 @@ var/global/list/image/splatter_cache=list()
...
-	var/obj/item/organ/external/r_foot = GET_EXTERNAL_ORGAN(perp, BP_R_FOOT)
+	var/mob/living/M = AM
+	var/obj/item/organ/external/l_foot = GET_EXTERNAL_ORGAN(M, BP_L_FOOT)
+	var/obj/item/organ/external/r_foot = GET_EXTERNAL_ORGAN(M, BP_R_FOOT)
 	var/hasfeet = l_foot && r_foot
 
 	var/transferred_data = blood_data ? blood_data[pick(blood_data)] : null
-	var/obj/item/clothing/shoes/shoes = perp.get_equipped_item(slot_shoes_str)
-	if(istype(shoes) && !perp.buckled)//Adding blood to shoes
+	var/obj/item/clothing/shoes/shoes = M.get_equipped_item(slot_shoes_str)
```
Comment (out-of-phaze) [inline code/game/objects/effects/decals/Cleanable/humans.dm:101]: Note to self: change this back to something other than `M` eventually.

File: code/game/objects/effects/mines.dm [hunk 3]
```diff
@@ -14,7 +14,8 @@
 	icon_state = "uglyminearmed"
 
 /obj/effect/mine/Crossed(atom/movable/AM)
-	Bumped(AM)
+	if(!isobserver(AM))
```
Comment (out-of-phaze) [inline code/game/objects/effects/mines.dm:17]: I think at some point this should be changed to a simulated check.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**